tests/php/src/Validation/ScannableURLProviderTest.php

```diff
@@ -0,0 +1,398 @@
+<?php
+
+namespace AmpProject\AmpWP\Tests\Validation;
+
+use AmpProject\AmpWP\Option;
+use AMP_Options_Manager;
+use AMP_Post_Meta_Box;
+use AMP_Theme_Support;
+use AmpProject\AmpWP\Tests\Helpers\AssertContainsCompatibility;
+use AmpProject\AmpWP\Tests\Helpers\PrivateAccess;
+use AmpProject\AmpWP\Tests\Helpers\ValidationRequestMocking;
+use AmpProject\AmpWP\Validation\ScannableURLProvider;
+use WP_Query;
+use WP_UnitTestCase;
+
+/** @coversDefaultClass \AmpProject\AmpWP\Validation\ScannableURLProvider */
+final class ScannableURLProviderTest extends WP_UnitTestCase {
+	use PrivateAccess, AssertContainsCompatibility;
+
+	/**
+	 * Validation URL provider instance to use.
+	 *
+	 * @var ScannableURLProvider
+	 */
+	private $validation_url_provider;
+
+	/**
+	 * Setup.
+	 *
+	 * @inheritdoc
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->validation_url_provider = new ScannableURLProvider();
+		add_filter( 'pre_http_request', [ ValidationRequestMocking::class, 'get_validate_response' ] );
+	}
+
+	/** @covers ::__construct */
```

Comment:
We don't want to commit perjury and upset the senate now :wink: 

```suggestion
	/** @covers ::__construct() */
```